### PR TITLE
For #24857: Add metrics for addresses autofill.

### PR DESCRIPTION
**Pull request by mcarare**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: .experimenter.yaml [hunk 1]
```diff
@@ -33,8 +33,8 @@ messaging:
       type: string
       description: What should be displayed when a control message is selected.
       enum:
-        - show-next-message
         - show-none
+        - show-next-message
```
Comment (mcarare) [inline .experimenter.yaml:37]: @jhugman Do you have any idea why this change keeps being made on app build? Should I commit this change to the main branch (on a separate commit/ PR), or is it something local and I should ignore it? Thank you!

**Comment by mcarare** [comment]
Needs. https://github.com/mozilla-mobile/android-components/pull/12152

**Comment by mcarare** [comment]
# Request for data collection review form

**All questions are mandatory. You must receive a review from a data steward peer on your responses to these questions before shipping new data collection.**

_1) What questions will you answer with this data?_

Are our users discovering the feature to use address autofill and are they finding it valuable?
How many of our users are filling addresses from autofill?

_2) Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?_
This is needed to analyze feature usage and possible feature improvements.

_3) What alternative methods did you consider to answer these questions? Why were they not sufficient?_
This is the only way.

_4) Can current instrumentation answer these questions?_
No. We need a new specific event for a specific app feature.

_5) List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Firefox/Data_Collection) found on the Mozilla wiki._

_**Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.**_

<table>
  <tr>
    <td>Measurement Description</td>
    <td>Data Collection Category</td>
    <td>Tracking Bug #</td>
  </tr>

  <tr>
    <td> An address autofill form was detected</td>
    <td>Category 2</td>
    <td>https://github.com/mozilla-mobile/fenix/issues/24857 </td>
    </tr>

  <tr>
    <td> An address autofill form was filled successfully </td>
    <td>Category 2</td>
    <td>https://github.com/mozilla-mobile/fenix/issues/24857 </td>
    </tr>

  <tr>
    <td> An address autofill prompt was displayed </td>
    <td>Category 2</td>
    <td>https://github.com/mozilla-mobile/fenix/issues/24857 </td>
    </tr>

  <tr>
    <td> An address autofill prompt was expanded </td>
    <td>Category 2</td>
    <td>https://github.com/mozilla-mobile/fenix/issues/24857 </td>
    </tr>

  <tr>
    <td> An address autofill prompt was dismissed </td>
    <td>Category 2</td>
    <td>https://github.com/mozilla-mobile/fenix/issues/24857 </td>
    </tr>

</table>

_6) How long will this data be collected?  Choose one of the following:_
Up until version 107, with the option to renew at that point. 

_7) What populations will you measure?_
All release channels and locales.

_8) If this data collection is default on, what is the opt-out mechanism for users?_
 Default Glean SDK opt-out mechanism. : Users can opt out of data collection by disabling "Usage and technical data" from Settings -> Data collection.

_9) Please provide a general description of how you will analyze this data._
 Glean 

_10) Where do you intend to share the results of your analysis?_
Only on Glean, mobile teams, and BD have internal access.

_12) Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?_
No.

**Review by Amejia481 — APPROVED** [review]

File: app/src/test/java/org/mozilla/fenix/settings/address/AddressEditorViewTest.kt [hunk 2]
```diff
@@ -14,19 +14,24 @@ import kotlinx.coroutines.runBlocking
...
 import org.junit.runner.RunWith
+import org.mozilla.fenix.GleanMetrics.Addresses
 import org.mozilla.fenix.R
 import org.mozilla.fenix.databinding.FragmentAddressEditorBinding
 import org.mozilla.fenix.helpers.FenixRobolectricTestRunner
 import org.mozilla.fenix.settings.address.interactor.AddressEditorInteractor
 import org.mozilla.fenix.settings.address.view.AddressEditorView
 
-@RunWith(FenixRobolectricTestRunner::class)
+@RunWith(FenixRobolectricTestRunner::class) // For gleanTestRule
```
[the remark this answers is not in the document]
Comment (mcarare) [reply app/src/test/java/org/mozilla/fenix/settings/address/AddressEditorViewTest.kt:34]: It is added in every other place we use it just for gleanTestRule and no other test features.

[the remark this answers is not in the document]
Comment (Amejia481) [reply app/src/test/java/org/mozilla/fenix/settings/address/AddressEditorViewTest.kt:34]: 👍🏽

**Comment by rocketsroger** [comment]
> # Request for data collection review form
> 
> **All questions are mandatory. You must receive a review from a data steward peer on your responses to these questions before shipping new data collection.**
> 
> _1) What questions will you answer with this data?_
> 
> Are our users discovering the feature to use address autofill and are they finding it valuable? How many of our users are filling addresses from autofill?
> 
> _2) Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements?_ This is needed to analyze feature usage and possible feature improvements.
> 
> _3) What alternative methods did you consider to answer these questions? Why were they not sufficient?_ This is the only way.
> 
> _4) Can current instrumentation answer these questions?_ No. We need a new specific event for a specific app feature.
> 
> _5) List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Firefox/Data_Collection) found on the Mozilla wiki._
> 
> _**Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.**_
> Measurement Description 	Data Collection Category 	Tracking Bug #
> An address autofill form was detected 	Category 2 	https://github.com/[/issues/24857](https://github.com/mozilla-mobile/fenix/issues/24857)
> An address autofill form was filled successfully 	Category 2 	https://github.com/[/issues/24857](https://github.com/mozilla-mobile/fenix/issues/24857)
> An address autofill prompt was displayed 	Category 2 	https://github.com/[/issues/24857](https://github.com/mozilla-mobile/fenix/issues/24857)
> An address autofill prompt was expanded 	Category 2 	https://github.com/[/issues/24857](https://github.com/mozilla-mobile/fenix/issues/24857)
> An address autofill prompt was dismissed 	Category 2 	https://github.com/[/issues/24857](https://github.com/mozilla-mobile/fenix/issues/24857)
> 
> _6) How long will this data be collected? Choose one of the following:_ Up until version 107, with the option to renew at that point. 
> 
> _7) What populations will you measure?_ All release channels and locales.
> 
> _8) If this data collection is default on, what is the opt-out mechanism for users?_ Default Glean SDK opt-out mechanism. : Users can opt out of data collection by disabling "Usage and technical data" from Settings -> Data collection.
> 
> _9) Please provide a general description of how you will analyze this data._ Glean
> 
> _10) Where do you intend to share the results of your analysis?_ Only on Glean, mobile teams, and BD have internal access.
> 
> _12) Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?_ No.

# Data Review

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way? 

Yes, through the metrics.yaml file and the [Glean Dictionary](https://dictionary.telemetry.mozilla.org/apps/fenix )

2) Is there a control mechanism that allows the user to turn the data collection on and off? 

Yes, through the "Send Usage Data" preference in the application settings

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

N/A, collection set to end or be renewed by version 107

4) Using the **[category system of data types](https://wiki.mozilla.org/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

Category 2, Interaction data

5) Is the data collection request for default-on or default-off?

default-on

6) Does the instrumentation include the addition of **any *new* identifiers**?

No

7) Is the data collection covered by the existing Firefox privacy notice? 

Yes

8) Does the data collection use a third-party collection tool? 

No

## Result

data-review+
